File: app/src/main/java/com/kelsos/mbrc/commands/model/UpdateLastFm.java
package com.kelsos.mbrc.commands.model;

import com.fasterxml.jackson.databind.node.BooleanNode;
import javax.inject.Inject;
import com.kelsos.mbrc.interfaces.ICommand;
import com.kelsos.mbrc.interfaces.IEvent;
import com.kelsos.mbrc.model.MainDataModel;

public class UpdateLastFm implements ICommand {
  private MainDataModel model;

  @Inject public UpdateLastFm(MainDataModel model) {
    this.model = model;
  }

  @Override public void execute(IEvent e) {
    model.setScrobbleState(((BooleanNode) e.getData()).asBoolean());
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/commands/model/UpdateLastFm.kt
package com.kelsos.mbrc.commands.model

import com.fasterxml.jackson.databind.node.BooleanNode
import com.kelsos.mbrc.events.bus.RxBus
import com.kelsos.mbrc.events.ui.ScrobbleChange
import com.kelsos.mbrc.interfaces.ICommand
import com.kelsos.mbrc.interfaces.IEvent
import com.kelsos.mbrc.model.MainDataModel
import javax.inject.Inject

class UpdateLastFm
@Inject
constructor(
  private val model: MainDataModel,
  private val bus: RxBus
) : ICommand {

  override fun execute(e: IEvent) {
    val newScrobbleState = (e.data as BooleanNode).asBoolean()
    if (newScrobbleState != model.isScrobblingEnabled) {
      model.isScrobblingEnabled = newScrobbleState
      bus.post(ScrobbleChange(newScrobbleState))
    }
  }
}
